mudslide/models.py

```diff
@@ -553,12 +562,205 @@ def dV(self, X: ArrayLike) -> ArrayLike:
         return out.reshape([5, 2, 2])
 
 
+class TMModel(AdiabaticModel_):
+    def __init__(
+        self,
+        turbomole_dir: str,
+        states: ArrayLike,
+        sub_dir_stem:str = "traj", 
+        representation: str = "adiabatic",
+        reference: Any = None,
+    ):
+        AdiabaticModel_.__init__(self, representation=representation, reference=reference)
+
+        self.turbomole_dir = turbomole_dir
+        self.states = states
+        self.nstates_ = len(self.states)
+        self.sub_dir_stem = sub_dir_stem
+        self.sub_dir_num = 0
+
+        self.turbomole_init()
+
+    def turbomole_init(self):
+        self.coord_path = Path(self.turbomole_dir)/"control"  
+        self.get_coords()
+        self.call_and_parse_turbomole()
+
+    def get_coords(self):
+        with open(self.coord_path, "r") as coord_file:
+            lines = coord_file.readlines()
+
+        coords = []
+        self.atom_order = []
+
+        try:
+            read_start = np.argwhere(np.array(["$coord\n" in line for line in lines]))[0][0]
```

Comment:
This is really wasteful. You're basically turning the whole thing into an array of `True`/`False` just to find the first line, and then only returning the line number. It will also fail if anything even slightly different is encounted. For example,
```
$coord ! this is an allowed comment by turbomole
```
would fail because there are things after the `d`.
```
 $coord
```
would fail because of the space. `$coord ` would fail because of the space at the end of the line. All of those are allowed by turbomole, so should be covered by us.

The best thing to do though is to use some built in facilities in turbomole. The `sdg` script/program will do most of what you need. For example,
```
sdg coord
```
will print out the coord block like
```
$coord
    2.68245142153191     -0.01532925226678      0.00000012933989  c
    5.35661968932518     -0.01570554780053     -0.00000007307270  c
   -1.37551527364443      2.32857147093840      0.00000011469638  c
   -2.71372989497353      4.64270033461115     -0.00000008888730  c
    5.16301691170948      8.97194268320440      0.00000011508832  c
    3.82944899733982      6.65643153409124     -0.00000006556916  c
   10.35925385641554     -0.01351664484525     -0.00000010603421  c
    7.68662189081884     -0.01396492275135      0.00000006587935  c
   -5.21925276592780      8.97196059766757     -0.00000010870875  c
   -3.88677960360603      6.65562142964378      0.00000006945337  c
    1.32870757012246      2.32575323529588     -0.00000011705168  c
    2.66497386930571      4.63899958287558      0.00000006486653  c
```
Running `sdg -f` will tell you just the filename that the data group is in. For example,
```
> sdg -f coord
coord
> sdg -f dft
control
```

Using the built-in functions is a much better idea because they are maintained by turbomole so they will better handle edge cases and things like that.

Reply:
Addressed!